### DISCUSSION TITLE
Multiple Columns in GGplot2

### DISCUSSION THREAD

**Discussion by eeinnew** [teamalgoritma/community#78] [Q&A] [answered]
Hello, I would like to ask about how to have 2 columns data in my plot.
I have several problems : 
1. I can not use geom_col as it stated that my data is discrete used in continuous variables, but when I use lolypop bar, it works (however, I want to use bar chart)
2. I want to plot Merchandise Value Exported and imported but do not know how, I try melt(based on my search on google) but seems do not work. 
3. I want to know on how to show different value in the glue function as I will be having 2 data here (export and import)
![WhatsApp Image 2022-05-05 at 15 06 40](https://user-images.githubusercontent.com/81936292/166884580-e5252ff0-41dc-44d3-a6d4-b8e36844d510.jpeg)

Thankyou in Advance!

**Comment by onlyphantom** [comment]
Hey Wennie, can you output a small sample of your dataset so we can reproduce it? Even the first 10 rows would work :)

**Comment by eeinnew** [comment]
Here you go! :)
[Business.csv](https://github.com/teamalgoritma/community/files/8629803/Business.csv)

**Comment by onlyphantom** [comment, marked as answer]
Take you for the sample data.

Make sure that your `Merchandise.exports.by.the.reporting.economy..current.US..` column is not read in as a character. This is due to the character strings containing comma. Hence when R reads `2,999` it won't read it as an integer or float but rather a character.

You can clean out the comma by doing this:
```
business$Merchandise.exports.by.the.reporting.economy..current.US.. <- as.numeric(gsub(",", "", business$Merchandise.exports.by.the.reporting.economy..current.US..))
```

Once you have that set up correctly, correct the ggplot code to take `label` out of `aes` mapping since `label` isn't the name of any column in your dataframe. 

The rest of the code can remain the same. 

```
ggplot(business, 
      aes(x=Time, 
          y=Merchandise.exports.by.the.reporting.economy..current.US..)) +
  geom_col(aes(fill=Merchandise.exports.by.the.reporting.economy..current.US..)) +
  labs(x="Year", y="Merchanise Exports") +
  theme(legend.position = "none")
```

Produces this:
![image](https://user-images.githubusercontent.com/16984453/167084260-2b73a199-2cbc-4ca7-b9f7-306d7a1c2fb2.png)

Sorry for the late reply -- let me know if this solves the issue for you? Otherwise happy to hop on a Meet call to walk you through it.

**Reply by eeinnew** [reply]
Well, Thankyou sir, it actually helps me with my first problem. However,  the second and the third are not settled yet. Anyway, meet call seems helpful, but how do we arrange it?

**Reply by onlyphantom** [reply]
> 2. I want to plot Merchandise Value Exported and imported but do not know how

In the code example I posted, I did use Merchandise Value Exported (`Merchandise.exports.by.the.reporting.economy..current.US..`). Does that not do what you want?

> 3. I want to know on how to show different value in the glue function as I will be having 2 data here (export and import)

I'm not fully certain I understand the question. 

Post a google meet link if you want and I'll hop in!

**Reply by onlyphantom** [reply]
Oh, upon re-reading it. I think this is what you're after:

```
library(reshape)
df <- melt(business[,c(3,6,7)], id=c("Time"))

ggplot(df, 
      aes(x=Time, 
          y=value,
          fill=variable)) +
  geom_col(position = "dodge") +
  labs(x="Year", y="Merchanise Value") +
  theme(legend.position = "bottom")
```

Produces:
![image](https://user-images.githubusercontent.com/16984453/167093726-fbeaa72a-50af-448d-b066-b985abcf529d.png)

If you need me to go through the code with you, feel free to send a Google Meet link.

**Comment by eeinnew** [comment]
okay, it is settled now! Thankyou!
